src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt

```diff
@@ -37,6 +38,7 @@ class KotlinterPlugin : Plugin<Project> {
 
                 val lintKotlin = registerParentLintTask()
                 val formatKotlin = registerParentFormatTask()
+                registerPrePushHookTask()
```

Comment:
You'll need to guard this with
```kotlin
if (project.rootProject == project) {
    registerPrePushHookTask()
}
```
Otherwise in multi-module builds the task will be registered on each module and running it from the top triggers multiple executions.

Reply:
Great catch